src/posit/connect/jobs.py

```diff
@@ -0,0 +1,273 @@
+
+        There is no implied ordering so if order matters, you should specify it yourself.
+
+        Parameters
+        ----------
+        id : str, not required
+            A unique identifier for the job.
+        ppid : Optional[str], not required
+            Identifier of the parent process.
+        pid : str, not required
+            Identifier of the process running the job.
+        key : str, not required
+            A unique key to identify this job.
+        remote_id : Optional[str], not required
+            Identifier for off-host execution configurations.
+        app_id : str, not required
+            Identifier of the parent content associated with the job.
+        variant_id : str, not required
+            Identifier of the variant responsible for the job.
+        bundle_id : str, not required
+            Identifier of the content bundle linked to the job.
+        start_time : str, not required
+            RFC3339 timestamp indicating when the job started.
+        end_time : Optional[str], not required
+            RFC3339 timestamp indicating when the job finished.
+        last_heartbeat_time : str, not required
+            RFC3339 timestamp of the last recorded activity for the job.
+        queued_time : Optional[str], not required
+            RFC3339 timestamp when the job was added to the queue.
+        status : int, not required
+            Current status. Options are 0 (Active), 1 (Finished), and 2 (Finalized)
+        exit_code : Optional[int], not required
+            The job's exit code, available after completion.
+        hostname : str, not required
+            Name of the node processing the job.
+        cluster : Optional[str], not required
+            Location where the job runs, either 'Local' or the cluster name.
+        image : Optional[str], not required
+            Location of the content in clustered environments.
+        run_as : str, not required
+            UNIX user responsible for executing the job.
+        queue_name : Optional[str], not required
+            Name of the queue processing the job, relevant for scheduled reports.
+        tag : JobTag, not required
+            A tag categorizing the job type. Options are build_jupyter, build_report, build_site, configure_report, git, packrat_restore, python_restore, render_shiny, run_api, run_app, run_bokeh_app, run_dash_app, run_fastapi_app, run_pyshiny_app, run_python_api, run_streamlit, run_tensorflow, run_voila_app, testing, unknown, val_py_ext_pkg, val_r_ext_pkg, and val_r_install.
+
+        Returns
+        -------
+        Optional[Job]
+        """
+        ...
+
+    @overload
+    def find_by(self, **conditions): ...
+
+    def find_by(self, **conditions) -> Optional[Job]:
+        return super().find_by(**conditions)
+
+
+class JobsMixin(Active, Resource):
+    """Mixin class to add a jobs attribute to a resource."""
+
+    def __init__(self, ctx, **kwargs):
```

Comment:
This feels like an abuse of inheritance to achieve execution. I'd argue the `Jobs(Job, ctx, self)` should be put into a function.

```python
def make_jobs(ctx: Context, parent: Active):
    return Jobs(Job, ctx, parent)
```